xarray/testing/assertions.py

```diff
@@ -50,21 +52,72 @@ def _data_allclose_or_equiv(arr1, arr2, rtol=1e-05, atol=1e-08, decode_bytes=Tru
 
 
 @ensure_warnings
-def assert_equal(a, b):
+def assert_isomorphic(a: DataTree, b: DataTree, from_root: bool = False):
+    """
+    Two DataTrees are considered isomorphic if every node has the same number of children.
+
+    Nothing about the data in each node is checked.
+
+    Isomorphism is a necessary condition for two trees to be used in a nodewise binary operation,
+    such as tree1 + tree2.
+
+    By default this function does not check any part of the tree above the given node.
+    Therefore this function can be used as default to check that two subtrees are isomorphic.
+
+    Parameters
+    ----------
+    a : DataTree
+        The first object to compare.
+    b : DataTree
+        The second object to compare.
+    from_root : bool, optional, default is False
+        Whether or not to first traverse to the root of the trees before checking for isomorphism.
+        If a & b have no parents then this has no effect.
+
+    See Also
+    --------
+    DataTree.isomorphic
+    assert_equal
+    assert_identical
+    """
+    __tracebackhide__ = True
+    assert isinstance(a, type(b))
+
+    if isinstance(a, DataTree):
+        if from_root:
+            a = a.root
+            b = b.root
+
+        assert a.isomorphic(b, from_root=from_root), diff_datatree_repr(
+            a, b, "isomorphic"
+        )
+    else:
+        raise TypeError(f"{type(a)} not of type DataTree")
+
+
+@ensure_warnings
+def assert_equal(a, b, **kwargs):
     """Like :py:func:`numpy.testing.assert_array_equal`, but for xarray
     objects.
 
     Raises an AssertionError if two objects are not equal. This will match
     data values, dimensions and coordinates, but not names or attributes
     (except for Dataset objects for which the variable names must match).
-    Arrays with NaN in the same location are considered equal.
+    Arrays with NaN in the same location are considered equal. DataTrees are
+    equal if they have isomorphic node structures, with matching node names,
+    and if they have matching variables and coordinates, all of which are
+    equal.
```

Comment:
The definition here of `equal/allclose/identical` is "isomorphic, and also every set of corresponding nodes in the trees are equal/allclose/identical". It would be good to reword this to indicate that `assert_equal` for trees is essentially just `assert_equal` for `Dataset` but mapped over the trees.